### ISSUE TITLE
Use of uninitialized value in numeric le (<=) at ExtUtils/Builder/Plan.pm line 52

### ISSUE THREAD

**Issue by skaji** [Leont/extutils-builder-plan#3]
I want to create a plan in ExtUtils::Builder::Plan
as written in Makefile syntax:

```Makefile
# Makefile

file1.txt: generate_files

file2.txt: generate_files

generate_files:
	echo foo > file1.txt
	echo bar > file2.txt

.PHONY: generate_files
```

So I wrote the following code:

```perl
# perl: test.pl

use strict;
use warnings;

use ExtUtils::Builder::Action::Command;
use ExtUtils::Builder::Node;
use ExtUtils::Builder::Planner;

my $planner = ExtUtils::Builder::Planner->new;

my $node1 = ExtUtils::Builder::Node->new(
    target => "file1.txt",
    dependencies => ["generate_files"],
);

my $node2 = ExtUtils::Builder::Node->new(
    target => "file2.txt",
    dependencies => ["generate_files"],
);

my $node3 = ExtUtils::Builder::Node->new(
    target => "generate_files",
    phony => 1,
    actions => [
        ExtUtils::Builder::Action::Command->new(
            command => ["bash", "-c", "echo foo > file1.txt; echo bar > file2.txt"]
        )
    ],
);

$planner->add_node($node1);
$planner->add_node($node2);
$planner->add_node($node3);

my $plan = $planner->materialize;

$plan->run( ["file1.txt"] );
```

Then, I got "Use of uninitialized value in numeric le" warning:
```
❯ perl test.pl
bash -c echo foo > file1.txt; echo bar > file2.txt
Use of uninitialized value in numeric le (<=) at /Users/skaji/env/perl/versions/5.40.0/lib/site_perl/5.40.0/ExtUtils/Builder/Plan.pm line 52.
```

Am I doing something wrong?

---

My environment
* macos 14.6.1, arm64
* perl 5.40.0
* ExtUtils::Builder::Plan 0.011

**Comment by Leont** [comment]
Generally speaking, it's bad design to depend on phony targets from non-phony targets because it will keep rebuilding them on every `make` invocation. I would strongly recommend against writing your build file this way (in `make` and in `ExtUtils::Builder`).

But you are right, this should work, and not give an error. Will look into fixing that.

**Comment by Leont** [comment]
As a side note, I would probably have written that as:

```perl
# perl: test.pl

use strict;
use warnings;

use ExtUtils::Builder::Action::Command;
use ExtUtils::Builder::Planner;

my $planner = ExtUtils::Builder::Planner->new;

$planner->create_node(
    target => "file1.txt",
    dependencies => ["generate_files"],
);

$planner->create_node(
    target => "file2.txt",
    dependencies => ["generate_files"],
);

$planner->create_node(
    target => "generate_files",
    phony => 1,
    actions => [
        ExtUtils::Builder::Action::Command->new(
            command => "echo foo > file1.txt; echo bar > file2.txt"
        )
    ],
);

my $plan = $planner->materialize;

$plan->run( "file1.txt" );
```
Using `create_node`, and the string form of command (which isn't well documented).

**Comment by Leont** [comment]
This should be fixed with [ExtUtils-Builder-0.012](https://metacpan.org/release/LEONT/ExtUtils-Builder-0.012)
